### ISSUE TITLE
Multiline comment causes indentation failure

### ISSUE THREAD

**Issue by ingydotnet** [jashkenas/coffeescript#4034]
```
> cat x.coffee 
class X
  foo: ->
###
stuff
###
  bar: ->
  baz: ->
> coffee -p x.coffee 
/home/ingy/x.coffee:7:10: error: unexpected indentation
  baz: ->
         ^
```

A few things:

* Indenting the `###` makes it parse (even if the content is not indented)
* Unindented single line comments work fine in the same context
* The error does not occur until the end of the stream

I was using `###` to comment out parts of a file that I was porting to CS, when I tripped on this.

**Comment by michaelficarra** [comment]
See duplicates.

**Comment by ingydotnet** [comment]
I looked for a duplicate, and found things similar but not the same. Can you point out a dup?

**Comment by michaelficarra** [comment]
This is probably our most duplicated issue. #3993, #3717, #3159, #2973, #2805, #2445, #2374, #2344, #2025, #1925, #1559, #1375.

**Comment by GeoffreyBooth** [comment]
This is fixed via #4572. Sorry it took awhile.
